app/controllers/concerns/foreman/controller/discovered_extensions.rb

```diff
@@ -48,6 +48,33 @@ def perform_auto_provision host, rule
     end
   end
 
+  def perform_reboot_all hosts = Host::Discovered.all
+    result = true
+    error_message = _("Errors during reboot: %s")
+    overall_errors = ""
+
+    if hosts.count > 0
+      Host.transaction do
+        hosts.each do |discovered_host|
+          begin
+            unless discovered_host.reboot
+              overall_errors << "#{discovered_host.name}: failed to reboot "
+            end
+          rescue Exception => e
+            overall_errors << "#{discovered_host.name}: #{e.to_s} "
+          end
+        end
+      end
+    else
+      error_message = _("No discovered hosts to reboot")
+      result = false
+    end
+
+    if overall_errors.present? || !result
+      error_message % overall_errors.strip
+    end
```

Comment:
Since you rely on the exit value, I'd prefer to be explicit: `else false`. Smells like place for regressions.